internal/kafka/command_topic_update.go

```diff
@@ -114,11 +116,39 @@ func (c *command) update(cmd *cobra.Command, args []string) error {
 		if err != nil {
 			return err
 		}
-		updateResp, err := kafkaREST.CloudClient.UpdateKafkaTopicPartitionCount(topicName, kafkarestv3.UpdatePartitionCountRequestData{PartitionsCount: int32(updateNumPartitionsInt)})
+		_, err = kafkaREST.CloudClient.UpdateKafkaTopicPartitionCount(topicName, kafkarestv3.UpdatePartitionCountRequestData{PartitionsCount: int32(updateNumPartitionsInt)})
 		if err != nil {
 			return err
 		}
-		configsValues[numPartitionsKey] = fmt.Sprint(updateResp.PartitionsCount)
+
+		errPartition := retry.Retry(time.Second/10, time.Second, func() error {
+			topic, httpRespPartition, err := kafkaREST.CloudClient.GetKafkaTopic(topicName)
+			if err != nil {
+				if restErr, parseErr := kafkarest.ParseOpenAPIErrorCloud(err); parseErr == nil && restErr.Code == ccloudv2.UnknownTopicOrPartitionErrorCode {
+					return fmt.Errorf(errors.UnknownTopicErrorMsg, topicName)
+				}
+				return kafkarest.NewError(kafkaREST.CloudClient.GetUrl(), err, httpRespPartition)
+			}
+
+			if int64(topic.PartitionsCount) != (updateNumPartitionsInt) {
+				return fmt.Errorf(`the partition count not yet updated`)
+			}
+			return nil
+		})
+
+		if errPartition != nil {
+			return errPartition
+		}
+
+		topic, httpRespPartition, errUpdated := kafkaREST.CloudClient.GetKafkaTopic(topicName)
+		if errUpdated != nil {
+			if restErr, parseErr := kafkarest.ParseOpenAPIErrorCloud(errUpdated); parseErr == nil && restErr.Code == ccloudv2.UnknownTopicOrPartitionErrorCode {
+				return fmt.Errorf(errors.UnknownTopicErrorMsg, topicName)
+			}
+			return kafkarest.NewError(kafkaREST.CloudClient.GetUrl(), errUpdated, httpRespPartition)
+		}
```

Comment:
Looks good, just one more suggestion: we can define a `topic` var outside of the retry loop and update it inside the retry loop, so that when it succeeds, we don't need to make one more API request.